### PR TITLE
Minio routes

### PR DESCRIPTION
**Pull request by yJulian**

### CODE REVIEW

File: src/routes/upload.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  Image.findOne({hash: hash}, (err, element) => {
+
+    if(element === undefined || element === null) { // No Element found, create new one
+      Image.create({type, hash}).then(value => {  // Create new Database entry
+        const _id = value._id as mongoose.Types.ObjectId; // Get created _id
+        os.bucketExists(bucketName).then(bucketExists => {  // Check if Minio Bucket exists
+          if ( !bucketExists ) {
+            os.makeBucket(bucketName, "", error => {  // Create Minio Bucket
+              if( error ) return console.log("error while creating bucket.", error);
```
Comment (Voakie) [inline src/routes/upload.ts:31]: Also respond with 500 here

File: src/routes/upload.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          } else {
+            os.putObject(bucketName, `${_id}.${type}`, data).then(r => {
+              res.json({id: _id});
+              res.end();
+            });
+          }
+        });
+      });
+    } else {
+      res.json({id: element._id});  //Return if from first entry with same hash
```
Comment (Voakie) [inline src/routes/upload.ts:46]: You forgot `res.end()`

File: src/main.ts [hunk 3]
```diff
@@ -10,9 +10,11 @@ import recipeRouter from "./routes/recipe";
 const app = express();
 const config = loadConfig();
 
+app.use(express.json());
 app.use(logger);
 app.use(cors());
 
+
 app.use("/recipe", recipeRouter);
```
Comment (Voakie) [inline src/main.ts:18]: You may want to add `app.use("/resources", uploadRoute)` (unless you planned something else)

**Review by Voakie — CHANGES_REQUESTED** [review]
Please also make sure to use [prettier](https://plugins.jetbrains.com/plugin/10456-prettier) for code formatting. Your formatting is all over the place right now 👀. If you have any questions regarding express or mongoose just ask me. Good work so far

File: src/routes/upload.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    if(!element) { // No Element found, create new one
+      Image.create({type, hash}).then(value => {  // Create new Database entry
+        const _id = value._id as mongoose.Types.ObjectId; // Get created _id
+        os.bucketExists(bucketName).then(bucketExists => {  // Check if Minio Bucket exists
+          if ( !bucketExists ) {
+            os.makeBucket(bucketName, "", error => {  // Create Minio Bucket
+              if( error ) {
+                res.status(500);
+                res.json({ error: "Internal Sever Error: Database", description: err });
+                res.end();
```
Comment (Voakie) [inline src/routes/upload.ts:39]: Should `return` after `res.end()` so nothing is put in minio

**Review by Voakie — CHANGES_REQUESTED** [review]
I hope I'm not annoying you with my requests 🥴

**Comment by yJulian** [comment]
> I hope I'm not annoying you with my requests 🥴

No everything fine 😊 I usually program in other languages and it's great that you can give me advises 

> Please also make sure to use [prettier](https://plugins.jetbrains.com/plugin/10456-prettier) for code formatting. Your formatting is all over the place right now 👀. If you have any questions regarding express or mongoose just ask me. Good work so far

IntelliJ dosn't show me a reformat with prettier option (Yes the plugin in installed)

File: src/routes/upload.ts (hunk 4 above)
Comment (yJulian) [reply src/routes/upload.ts:39]: Should be solved by the next push

**Comment by Voakie** [comment]
> IntelliJ dosn't show me a reformat with prettier option (Yes the plugin in installed)

Make sure you have it set up like me. Then it should work by just pressing `Strg + Alt + L`
image

**Review by Voakie — APPROVED** [review]
Good work 😎 keep it up

**Review by DeveloperTK — APPROVED** [review]
Y    E    S

**Pull request merged**